File: selenium-java/src/test/java/frameworks/seleniumjava/utils/TestApi.java
package frameworks.seleniumjava.utils;

import java.util.List;
import java.util.Map;

public final class TestApi {
  public static final Map<String, Object> DEMO_USER = Map.of(
      "id", 101,
      "username", "demo",
      "role", "qa-engineer",
      "features", List.of("ui", "api", "reporting")
  );

  private TestApi() {}
}

File: playwright-typescript/utils/testApi.ts
export type DemoUser = {
  id: number;
  username: string;
  role: string;
  features: string[];
};

export type ContactRequest = {
  email?: string;
  message?: string;
};

export const DEMO_USER: DemoUser = {
  id: 101,
  username: 'demo',
  role: 'qa-engineer',
  features: ['ui', 'api', 'reporting']
};

const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

export function validateContactRequest(payload: ContactRequest): { valid: boolean; error?: string } {
  const email = payload.email?.trim() ?? '';
  const message = payload.message?.trim() ?? '';

  if (!emailPattern.test(email)) {
    return { valid: false, error: 'A valid email is required.' };
  }

  if (message.length < 10) {
    return { valid: false, error: 'Message must be at least 10 characters long.' };
  }

  return { valid: true };
}
